### PR TITLE
NAS-101272: synchronize Advanced Replication Fields with docs

### PR DESCRIPTION
**Pull request by Mrt134**
- Update line break spacing
- Unify the text between UI and guide EDIT: https://github.com/freenas/freenas-docs/pull/1543

### CODE REVIEW

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 1]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
- replicated. All snapshots are replicated by default. To choose which\
- snapshots are replicated, set the checkbox and select a schedule from\
- the drop-down menu. For example, the periodic snapshot task takes a\
- snapshot every hour, but only every other snapshot is needed for\
- replication. The scheduler is set to even hours and only snapshots\
+    restrict_schedule_tooltip: T('Schedule which periodic snapshots are \
+ replicated. All snapshots are replicated by default. To choose which \
+ snapshots are replicated, set the checkbox and select a schedule from \
+ the drop-down menu. For example, the periodic snapshot task takes a \
+ snapshot every hour, but only every other snapshot is needed for \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]: This seems to be a specific example, but reads like a statement of fact: "only every other snapshot is needed". Reword to clarify that this is an example.  "As an example, consider a system which takes a snapshot every hour, but only replication of every other snapshot is desired."

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 2]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+ replication. The scheduler is set to even hours and only snapshots \
  taken at those times are replicated.'),
 
-    restrict_schedule_picker_tooltip: T('Select a preset schedule or\
- choose <i>Custom</i> to use the advanced scheduler. Spaces are not\
- allowed in <b>Minutes</b>, <b>Hours</b>, or <b>Days</b> of the custom\
+    restrict_schedule_picker_tooltip: T('Select a preset schedule or \
+ choose <i>Custom</i> to use the advanced scheduler. Spaces are not \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]: Again with the spaces.  If the custom scheduler still allows them, we need to enter a ticket to fix that.

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 3]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+ allowed in <b>Minutes</b>, <b>Hours</b>, or <b>Days</b> of the custom \
  scheduler.'),
 
     restrict_schedule_begin_placeholder: T('Begin'),
-    restrict_schedule_begin_tooltip: T('Define a starting time when the\
- replication cannot run. A replication that is in progress can continue\
+    restrict_schedule_begin_tooltip: T('Define a starting time when the \
+ replication cannot run. A replication that is in progress can continue \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
 replication is not allowed to start. A replication that is already in progress can continue \
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 4]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
  to run past this time.'),
 
     restrict_schedule_end_placeholder: T('End'),
-    restrict_schedule_end_tooltip: T('Define an ending time for the\
+    restrict_schedule_end_tooltip: T('Define an ending time for the \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
    restrict_schedule_end_tooltip: T('Set an ending time for when \
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 5]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
  restriction on activating replication schedules.'),
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
 replications are not allowed to start.'),
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 6]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
-    retention_policy_tooltip: T('When replicated snapshots are deleted\
- from the destination system:<ul>\
- <li><i>Same as Source</i>: duplicate the <b>Snapshot Lifetime</b>\
- setting from the source system.</li>\
- <li><i>Custom</i>: define a <b>Snapshot Lifetime</b> for the\
- destination system.</li>\
- <li><i>None</i>: never delete snapshots from the destination\
+    retention_policy_tooltip: T('When replicated snapshots are deleted \
+ from the destination system:<ul> \
+ <li><i>Same as Source</i>: duplicate the <b>Snapshot Lifetime</b> \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
 <li><i>Same as Source</i>: use the <b>Snapshot Lifetime</b> \
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 7]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+ setting from the source system.</li> \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
 from the source system.</li> \
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 8]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+    lifetime_value_tooltip: T('How long a snapshot remains on the \
+ destination system. Enter a number and choose a measure of time from \
  the drop-down.'),
 
     lifetime_unit_placeholder: T(''),
     lifetime_unit_tooltip: T(''),
 
     compression_placeholder: T('Stream Compression'),
-    compression_tooltip: T('Select a compression algorithm to reduce the\
+    compression_tooltip: T('Select a compression algorithm to reduce the \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
    compression_tooltip: T('Select a compression algorithm to use on the \
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 9]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
  size of the data being replicated.'),
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
 data being replicated.'),
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 10]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+    dedup_tooltip: T('Deduplicate the stream to avoid sending redundant \
+ data blocks. The destination system must also support deduplicated \
+ streams. See <a href="https://www.freebsd.org/cgi/man.cgi?query=zfs" \
  target="_blank">zfs(8)</a>.'),
 
     large_block_placeholder: T('Allow Blocks Larger than 128KB'),
-    large_block_tooltip: T('Enable the stream to send large data blocks.\
- The destination system must also support large blocks. See\
- <a href="https://www.freebsd.org/cgi/man.cgi?query=zfs"\
+    large_block_tooltip: T('Enable the stream to send large data blocks. \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]:
```suggestion
    large_block_tooltip: T('Allow sending large data blocks. \
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 11]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+    name_tooltip: T('Descriptive name for the replication. Replication \
+ task names must be unique.'),
 
     direction_placeholder: T('Direction'),
-    direction_tooltip: T('Direction of travel. <i>Push</i> sends\
- snapshots to a destination system. <i>Pull</i> receives snapshots from\
- a destination system.'),
+    direction_tooltip: T('Direction of travel. <i>Push</i> sends \
+ snapshots to a destination system. <i>Pull</i> receives snapshots from \
+ a destination system. Choosing <i>Pull</i> requires entering a snapshot \
```
[the remark this answers is not in the document]
Comment (Mrt134) [reply src/app/helptext/task-calendar/replication/replication.ts]: Push also supports local > local replication. Updating text to include that scenario as well.

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 12]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
-    ssh_credentials_tooltip: T('Choose the\
- <a href="--docurl--/system.html#ssh-connection"\
+    ssh_credentials_tooltip: T('Choose the \
+ <a href="--docurl--/system.html#ssh-connection" \
  target="_blank">SSH connection</a>.'),
 
     netcat_active_side_placeholder: T('Netcat Active Side'),
-    netcat_active_side_tooltip: T('Choose a system (<i>LOCAL</i> or\
- <i>REMOTE</i>) to open TCP ports and allow the connection between both\
+    netcat_active_side_tooltip: T('Choose a system (<i>LOCAL</i> or \
```
[the remark this answers is not in the document]
Comment (Mrt134) [reply src/app/helptext/task-calendar/replication/replication.ts]: Updated with draft replacement text

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 13]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
  datasets are on a remote system.'),
 
     target_dataset_placeholder: T('Target Dataset'),
-    target_dataset_tooltip: T('Choose a dataset on the destination\
- system where snapshots are stored. Click the <b>Folder icon</b> to see\
- all datasets on the destination system and click on a dataset to set it\
- as the target. An SSH connection must be selected for the browser to\
- display datasets from a remote system.'),
+    target_dataset_tooltip: T('Choose a dataset on the destination \
+ system where snapshots are stored. Click the <b>Folder icon</b> to see \
```
[the remark this answers is not in the document]
Comment (Mrt134) [reply src/app/helptext/task-calendar/replication/replication.ts]: Those don't exist in the webui. Please open a ticket for the UI team to add character replacements to the webui help text system.

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 14]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
     name_placeholder: T('Name'),
-    name_tooltip: T('Replication task name.'),
+    name_tooltip: T('Descriptive name for the replication.'),
 
     direction_placeholder: T('Direction'),
-    direction_tooltip: T('Direction of travel. <i>Push</i> sends\
- snapshots to a destination system. <i>Pull</i> receives snapshots from\
- a destination system.'),
+    direction_tooltip: T('Direction of travel. <i>Push</i> sends \
+ snapshots from the local system to either another dataset on the local \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]: I feel like this overcomplicates this sentence and makes it more difficult to understand for the standard use case. This is a little clearer:
```
<i>Push</i> sends snapshots from the local system to a remote system, or to another dataset on the local system.
```
Best to fix the UI form so it does not show Direction when local->local is chosen. Please enter a ticket for that.

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 15]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+ replication task. Choose from configured \
+ <a href="--docurl--/tasks.html#periodic-snapshot-tasks" \
+ target="_blank">Periodic Snapshot Tasks</a>. This replication task must \
+ have the same <b>Recursive</b> and <b>Exclude Child Datasets</b> values \
+ as the chosen periodic snapshot task. Selecting a periodic snapshot \
+ schedule removes the <b>Schedule</b> field.'),
 
     naming_schema_placeholder: T('Naming Schema'),
-    naming_schema_tooltip: T(''),
+    naming_schema_tooltip: T('Custom snapshots to be replicated. Enter \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]: This is not the actual snapshots, but a naming pattern that matches the snapshots. We did this more clearly on the other helptext: "Pattern that matches the names of snapshots to be replicated." (or something like that).

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 16]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+ <i>%m</i>, <i>%d</i>, <i>%H</i>, and <i>%M</i> string requirements as \
+ the <b>Naming Schema</b> in a \
+ <a href="--docurl--/tasks.html#periodic-snapshot-tasks" \
  target="_blank">periodic snapshot task.'),
 
     auto_placeholder: T('Run Automatically'),
-    auto_tooltip: T('Set to either start this replication task\
- immediately after the linked periodic snapshot task completes or see\
+    auto_tooltip: T('Set to either start this replication task \
+ immediately after the linked periodic snapshot task completes or see \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]: "or see" is not clear whether setting this shows options or the reader is supposed to go look at them.

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 17]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
  options to create a separate <b>Schedule</b> for this replication.'),
 
     schedule_placeholder: T('Schedule'),
-    schedule_tooltip: T('Define specific times to start snapshotting the\
- <b>Source Datasets</b>. Disables running the replication immediately\
- after the periodic snapshot task. Adds the <b>Begin</b> and <b>End</b>\
- fields.'),
+    schedule_tooltip: T('Define a specific time to start the replication \
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]: No, not define.  "Define a time" does not make sense.  The user sets or chooses or selects. But those are not needed here, we can just say what this field is:
```suggestion
    schedule_tooltip: T('Start time for the replication \
```

File: src/app/helptext/task-calendar/replication/replication.ts [hunk 18]
```diff
@@ -2,234 +2,243 @@ import { T } from '../../../translate-marker';
...
+ task.'),
 
-    schedule_picker_tooltip: T('Select a preset schedule or choose\
- <i>Custom</i> to use the advanced scheduler. Spaces are not allowed in\
- <b>Minutes</b>, <b>Hours</b>, or <b>Days</b> of the custom scheduler.'),
+    schedule_picker_tooltip: T('Select a preset schedule or choose \
+ <i>Custom</i> to use the advanced scheduler.'),
 
     schedule_begin_placeholder: T('Begin'),
```
Comment (wblock) [inline src/app/helptext/task-calendar/replication/replication.ts]: Not clear how this is different from the one above, which is just "schedule_tooltip".

File: src/app/helptext/task-calendar/replication/replication.ts (hunk 14 above)
Comment (Mrt134) [reply src/app/helptext/task-calendar/replication/replication.ts]: https://jira.ixsystems.com/browse/NAS-103227

File: src/app/helptext/task-calendar/replication/replication.ts (hunk 18 above)
Comment (Mrt134) [reply src/app/helptext/task-calendar/replication/replication.ts]: one is the checkbox to enable the field, the other is the field itself

**Pull request merged**